python/pdstools/reports/ModelReport.qmd

```diff
@@ -197,65 +328,135 @@ predictors. This can be configured via a parameter to this report.
-    display(Markdown(f"| Percentage Residual values | TODO |"))
+    base_propensity = (
+        predictor_data
+        .select(pl.last("Positives")/pl.last("ResponseCount"))
+        .collect()
+        .item()
+    )
 
-    # TODO predictor groups missing https://github.com/pegasystems/pega-datascientist-tools/issues/127
-    display(Markdown(f"| Predictor Group | TODO |"))
-    display(Markdown(f"| Correlated Predictors | TODO |"))
+    predictor_properties = (predictor_data
+        .select(
+            pl.last("Type"),
+            pl.last("Performance").round(4).alias("Univariate Performance"),
+            pl.last("EntryType").alias("Status"),
+            pl.format("{}%", ((pl.col("BinResponseCount").where(pl.col("BinType") == "MISSING").sum()) * 100 / pl.last("ResponseCount")).round(2)).alias("Missing"),
+            # Residual aka Symbolic Overflow only applies to symbolics. Interestingly
+            # there almost always is a residual bin. We don't know exactly how many
+            # items are in that bin unfortunately, and there is no exact way to derive.
+            pl.format("{}%", ((pl.col("BinResponseCount").where(pl.col("BinType") == "RESIDUAL").sum()) * 100 / pl.last("ResponseCount")).round(2)).alias("Residual"),
+
+            # # TODO predictor groups missing https://github.com/pegasystems/pega-datascientist-tools/issues/127
+            # display(Markdown(f"| Predictor Group | TODO |"))
+            # display(Markdown(f"| Correlated Predictors | TODO |"))
+        )
+    )
 
-    display(Markdown(": A Great Predictor {.striped .hover}"))
-    
-    # In the R version I managed to use a table to format it so
-    # the description is on the left, the plots on the right. With
-    # quarto I couldnt figure out, so the plots are seperate. Also
-    # the width/height of the figures doesnt seem changeable,
-    # show(width=20, height=10) has no effect
+    show(predictor_properties.collect().to_pandas())
 
     fig = datamart.plotPredictorBinning(modelids=[model_id], predictors=[pred])
 
     # Customize some of the default styling of the plot
-    # TODO except for title consider moving into library
-    # TODO make the fly-over better
-    fig.update_layout(title=pred, xaxis_title="")
+
+    fig.update_layout(title="Predictor Binning", xaxis_title="")
+    # fig.layout.yaxis2.tickformat = ",.3%"
     fig.data[0].opacity=0.5
     fig.data[1].line.color="#EF8B08"
     fig.data[1].line.width=3
     fig.data[1].marker.color="black"
+
+    # Add line for base propensity, TODO consider putting back in the library
+    fig.add_hline(y=base_propensity, line_dash="dash", line_color="grey", yref='y2')
+    # fig.update_xaxes(type='category') # prevent plotly from trying to guess
+
+    fig.update_layout(width=700, height=300)
+    fig.show()
+
+    # "Philip Mann" plot with simple red/green lift bars relative to base propensity
+    # TODO move below "Philip Mann" plot into a plot function (plotBinningLift)
```

Comment:
Highlighting as not to forget

Reply:
The plot functions are a little daunting, I'll first get the quarto feature complete then move things around